### ISSUE TITLE
HA integration?

### ISSUE THREAD

**Issue by andriej** [syssi/xiaomi_fan#58]
Any chance of this component becoming part of official HA integration?
Best domain, well known brand for other integrations, python-miio library....

Hope to see it there!

**Comment by insajd** [comment]
It needs some refactoring before being accepted as official.
See more info here, https://github.com/home-assistant/core/pull/20089

**Comment by HydrelioxGitHub** [comment]
Any news or progress on this ?

**Comment by keirans** [comment]
HA has just released the [xiaomi miio](https://github.com/home-assistant/core/tree/dev/homeassistant/components/xiaomi_miio) which looks like it could be the best place to transition this work over to now.

At the moment, it's a bit out of my skills range, however I am more than happy to help anyone with code reviews and testing.

I've got 2 fans that work OK under this component at the moment zhimi.fan.za4 & dmaker.fan.p5

Anyone who wants to pick this up, please do let me know. I have a feeling that this custom component will break soon as everything transitions over to xiaomi_miio and the fan support changes.

**Comment by syssi** [comment]
Correct. ;-) This is the first step. As soon it gets merged we can push the official Xiaomi Fan support.

**Comment by keirans** [comment]
Exciting times @syssi !

More than happy to help test all the new functionality for you if you need a hand with models you don't have.

**Comment by TheZoker** [comment]
Any news on this? 🙂

**Comment by xekil** [comment]
Any news on this?

**Comment by syssi** [comment]
The official `xiaomi_miio` component is heavy refactored at the moment. The next HA releases will ship some breaking changes. Attributes will be removed and moved to sensors. This is the fundation to gain speed and have official support soon again. :-)

**Comment by xekil** [comment]
ok, thank you
